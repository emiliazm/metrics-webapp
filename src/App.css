* {
  padding: 0;
  margin: 0;
  font-family: "Gill Sans MT", sans-serif;
}

img {
  width: 50px;
}
