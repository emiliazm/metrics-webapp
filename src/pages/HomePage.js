import React from 'react';
import ContinentsList from '../components/ContinentsList';
import CountriesList from '../components/CountriesList';

const Home = () => (
  <div className="container">
    <ContinentsList />
    <CountriesList />
  </div>
);

export default Home;
