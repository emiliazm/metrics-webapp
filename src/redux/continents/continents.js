import { getContinentsApi, getContinentByNameApi } from '../../api/requestApi';

const GET_CONTINENT = 'metricswebapp/continents/GET_CONTINENT';
const GET_CONTINENTS = 'metricswebapp/continents/GET_CONTINENTS';

const initialState = [];

export const getContinents = () => async (dispatch) => {
  const data = await getContinentsApi();
  const continents = data.map((continent) => {
    const {
      continent: continentName,
      cases,
      countries,
    } = continent;

    return {
      continentName,
      cases,
      countries,
    };
  });
  dispatch({ type: GET_CONTINENTS, continents });
};

export const getContinent = (continentName) => async (dispatch) => {
  const data = await getContinentByNameApi(continentName);
  dispatch({ type: GET_CONTINENT, data });
};

export const reducerContinents = (state = initialState, action) => {
  switch (action.type) {
    case GET_CONTINENTS:
      return action.continents;

    default:
      return state;
  }
};

export const reducerContinent = (state = initialState, action) => {
  switch (action.type) {
    case GET_CONTINENT:
      return action.data;

    default:
      return state;
  }
};
