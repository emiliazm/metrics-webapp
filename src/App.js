import {
  HashRouter as Router, Navigate, Route, Routes,
} from 'react-router-dom';
import './App.css';
import HomePage from './pages/HomePage';
import DetailsPage from './pages/DetailsPage';

function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="*" element={<Navigate to="/" />} />
        <Route path="/details" element={<DetailsPage />} />
      </Routes>
    </Router>
  );
}

export default App;
