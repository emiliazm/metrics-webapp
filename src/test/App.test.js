import { configureStore } from '@reduxjs/toolkit';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import App from '../App';
import { reducerContinent, reducerContinents } from '../redux/continents/continents';
import countriesReducer from '../redux/countries/countries';
import HomePage from '../pages/HomePage';

const setupStore = () => {
  const preloadedState = {
    continents: [
      {
        continentName: 'South America',
        cases: 59063198,
        countries: [
          'Argentina',
          'Bolivia',
          'Brazil',
          'Chile',
          'Colombia',
          'Ecuador',
          'Falkland Islands (Malvinas)',
          'French Guiana',
          'Guyana',
          'Paraguay',
          'Peru',
          'Suriname',
          'Uruguay',
          'Venezuela',
        ],
      },
    ],

    continent: [
      {
        active: 1725448,
        activePerOneMillion: 3941.64,
        cases: 59063198,
        casesPerOneMillion: 134924.95,
        continent: 'South America',
        continentInfo: { lat: -15.6551563, long: -100.7484231 },
        countries: [
          'Argentina',
          'Bolivia',
          'Brazil', 'Chile',
          'Colombia',
          'Ecuador',
          'Falkland Islands (Malvinas)',
          'French Guiana',
          'Guyana',
          'Paraguay',
          'Peru',
          'Suriname',
          'Uruguay',
          'Venezuela'],
        critical: 10371,
        criticalPerOneMillion: 23.69,
        deaths: 1302803,
        deathsPerOneMillion: 2976.14,
        population: 437748515,
        recovered: 56034947,
        recoveredPerOneMillion: 128007.17,
        tests: 224883556,
        testsPerOneMillion: 513727.74,
        todayCases: 76598,
        todayDeaths: 104,
        todayRecovered: 11119,
        updated: 1655935422476,
      },
    ],

    countries: [
      {
        countryName: 'Argentina',
        cases: 9341492,
        countryInfo: {
          flag: 'https://disease.sh/assets/img/flags/ar.png',
          _id: 32,
          iso2: 'AR',
          iso3: 'ARG',
          lat: -34,
          long: -64,
        },
        continent: 'South America',
      },
    ],
  };

  return configureStore({
    preloadedState,
    reducer: {
      continents: reducerContinents,
      continent: reducerContinent,
      countries: countriesReducer,
    },
  });
};

describe('Test the App components', () => {
  it('renders the App correctly', () => {
    const store = setupStore();
    const app = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });

  it('renders the Home page correctly', () => {
    const store = setupStore();
    const page = render(
      <Provider store={store}>
        <Router>
          <HomePage />
        </Router>
      </Provider>,
    );
    expect(page).toMatchSnapshot();
  });

  it('shows the corresponding page when clicking the continents links', () => {
    const store = setupStore();
    const { container } = render(
      <Provider store={store}>
        <App url="/" />
      </Provider>,
    );
    fireEvent.click(screen.getByText('South America'));
    expect(container.getElementsByClassName('container-countries')).toHaveLength(1);
  });

  it('shows the Home page when clicking the back link', () => {
    const store = setupStore();
    const { container } = render(
      <Provider store={store}>
        <App url="/" />
      </Provider>,
    );
    fireEvent.click(screen.getByText('Back'));
    expect(container.getElementsByClassName('container')).toHaveLength(1);
  });
});
