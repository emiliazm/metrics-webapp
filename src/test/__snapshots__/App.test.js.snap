// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the App components renders the App correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div>
          <nav
            class="nav-container"
          >
            <h1>
              Covid Cases by Continets
            </h1>
            <div
              class="icons-container"
            >
              <i
                class="material-symbols-outlined"
              >
                mic
              </i>
              <i
                class="material-symbols-outlined"
              >
                settings
              </i>
            </div>
          </nav>
          <div
            class="map"
          >
            <img
              alt="logo"
              class="nav-logo"
              src="map.png"
            />
          </div>
        </div>
        <div>
          <input
            placeholder="Search by continent here"
            type="text"
          />
        </div>
        <ul
          class="continents-ul"
        >
          <li
            class="continents-li"
          >
            <div>
              <i
                class="circle-arrow material-symbols-outlined"
              >
                arrow_circle_right
              </i>
            </div>
            <a
              href="#/South America"
            >
              <h2>
                South America
              </h2>
              <p>
                59063198
              </p>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div>
        <nav
          class="nav-container"
        >
          <h1>
            Covid Cases by Continets
          </h1>
          <div
            class="icons-container"
          >
            <i
              class="material-symbols-outlined"
            >
              mic
            </i>
            <i
              class="material-symbols-outlined"
            >
              settings
            </i>
          </div>
        </nav>
        <div
          class="map"
        >
          <img
            alt="logo"
            class="nav-logo"
            src="map.png"
          />
        </div>
      </div>
      <div>
        <input
          placeholder="Search by continent here"
          type="text"
        />
      </div>
      <ul
        class="continents-ul"
      >
        <li
          class="continents-li"
        >
          <div>
            <i
              class="circle-arrow material-symbols-outlined"
            >
              arrow_circle_right
            </i>
          </div>
          <a
            href="#/South America"
          >
            <h2>
              South America
            </h2>
            <p>
              59063198
            </p>
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test the App components renders the Home page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div>
          <nav
            class="nav-container"
          >
            <h1>
              Covid Cases by Continets
            </h1>
            <div
              class="icons-container"
            >
              <i
                class="material-symbols-outlined"
              >
                mic
              </i>
              <i
                class="material-symbols-outlined"
              >
                settings
              </i>
            </div>
          </nav>
          <div
            class="map"
          >
            <img
              alt="logo"
              class="nav-logo"
              src="map.png"
            />
          </div>
        </div>
        <div>
          <input
            placeholder="Search by continent here"
            type="text"
          />
        </div>
        <ul
          class="continents-ul"
        >
          <li
            class="continents-li"
          >
            <div>
              <i
                class="circle-arrow material-symbols-outlined"
              >
                arrow_circle_right
              </i>
            </div>
            <a
              href="#/South America"
            >
              <h2>
                South America
              </h2>
              <p>
                59063198
              </p>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div>
        <nav
          class="nav-container"
        >
          <h1>
            Covid Cases by Continets
          </h1>
          <div
            class="icons-container"
          >
            <i
              class="material-symbols-outlined"
            >
              mic
            </i>
            <i
              class="material-symbols-outlined"
            >
              settings
            </i>
          </div>
        </nav>
        <div
          class="map"
        >
          <img
            alt="logo"
            class="nav-logo"
            src="map.png"
          />
        </div>
      </div>
      <div>
        <input
          placeholder="Search by continent here"
          type="text"
        />
      </div>
      <ul
        class="continents-ul"
      >
        <li
          class="continents-li"
        >
          <div>
            <i
              class="circle-arrow material-symbols-outlined"
            >
              arrow_circle_right
            </i>
          </div>
          <a
            href="#/South America"
          >
            <h2>
              South America
            </h2>
            <p>
              59063198
            </p>
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
